src/pages/join.js

```diff
@@ -70,98 +76,120 @@ const Join = ({
       : intl.formatMessage(messages.feedbackChannelTitle)
 
   return (
-    <StudentLayout
-      isInteractionEnabled={isConfusionBarometerActive || isFeedbackChannelActive}
-      pageTitle={`Join ${shortname}`}
-      sidebar={{
-        activeItem: sidebarActiveItem,
-        handleSidebarActiveItemChange,
-        handleToggleSidebarVisible,
-        sidebarVisible,
-      }}
-      title={title}
-    >
-      <div className="joinSession">
-        {activeInstances.length > 0 ? (
-          <QuestionArea
-            active={sidebarActiveItem === 'activeQuestion'}
-            handleNewResponse={handleNewResponse}
-            questions={activeInstances}
-            sessionId={id}
-            shortname={shortname}
-          />
-        ) : (
-          <div
-            className={classNames('questionArea', {
-              inactive: sidebarActiveItem !== 'activeQuestion',
-            })}
-          >
-            <FormattedMessage defaultMessage="No question active." id="joinSession.noQuestionActive" />
-          </div>
-        )}
+    <Query query={JoinSessionQuery} variables={{ shortname }}>
+      {({ subscribeToMore }) => (
+        <StudentLayout
+          isInteractionEnabled={isConfusionBarometerActive || isFeedbackChannelActive}
+          pageTitle={`Join ${shortname}`}
+          sidebar={{
+            activeItem: sidebarActiveItem,
+            handleSidebarActiveItemChange,
+            handleToggleSidebarVisible,
+            sidebarVisible,
+          }}
+          subscribeToMore={() =>
+            subscribeToMore({
+              document: UpdateSessionSubscription,
+              updateQuery: (prev, { subscriptionData }) => {
+                if (!subscriptionData.data) return prev
+                const newFeedItem = subscriptionData.data.sessionUpdateAdded
+                return Object.assign({}, prev, {
+                  joinSession: {
+                    activeInstances: [newFeedItem.activeInstances, ...prev.joinSession.activeInstances],
+                    feedbacks: [newFeedItem.feedbacks, ...prev.joinSession.feedbacks],
```

Comment:
The feedbacks should never be changed by the updated session state, right? I think this could be removed.